tools/docs/generate.go

```diff
@@ -49,21 +49,55 @@ func markdown(path string, cmd *cobra.Command) {
 	must(doc.GenMarkdownTree(cmd, path))
 }
 
+func man(path string, header *doc.GenManHeader, cmd *cobra.Command) {
+	must(os.MkdirAll(path, 0755))
+	must(doc.GenManTree(cmd, header, path))
+}
+
+type commandWithManHeader struct {
+	command *cobra.Command
+	header  *doc.GenManHeader
+}
+
 func main() {
 	prefix := GetenvOr("DESTDIR", ".")
 	format := GetenvOr("FORMAT", "markdown")
 
-	apps := map[string]*cobra.Command{
-		path.Join(prefix, "kuma-cp"):            kuma_cp.DefaultRootCmd(),
-		path.Join(prefix, "kumactl"):            kumactl.DefaultRootCmd(),
-		path.Join(prefix, "kuma-dp"):            kuma_dp.DefaultRootCmd(),
-		path.Join(prefix, "kuma-prometheus-sd"): kuma_prometheus_sd.DefaultRootCmd(),
+	apps := map[string]commandWithManHeader{
+		path.Join(prefix, "kuma-cp"): commandWithManHeader{
+			command: kuma_cp.DefaultRootCmd(),
+			header: &doc.GenManHeader{
+				Title: "kuma-cp manual",
```

Comment:
For all these, let's fill in all the fields that make sense:

```Go
Title: "KUMA-CP" // note the uppercase
Section: "8", // IMHO kuma-dp and kuma-cp would be section 8, kumactl in section 1
Source:  fmt.Sprintf("%s %s", version.Product, version.Build.Version) // use pkg/version
Manual: version.Product,
```

We might need to tweak the makefile to set the right Go build tags to get the version info populated, but I think it would be cool if the version was in there (systemd does this). If it's too painful, then the source could be just the product name.

check the reference at https://man7.org/linux/man-pages/man7/man-pages.7.html

Reply:
addressed